### PR TITLE
Small refactor to  PlanFragment 

### PR DESCRIPTION
**Pull request by arhimondr**

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java [hunk 1]
```diff
@@ -199,17 +199,17 @@ private static void findRemoteSourceNodes(PlanNode node, Builder<RemoteSourceNod
 
     public PlanFragment withBucketToPartition(Optional<int[]> bucketToPartition)
     {
-        return new PlanFragment(id, root, symbols, partitioning, tableScanSchedulingOrder, partitioningScheme.withBucketToPartition(bucketToPartition), stageExecutionDescriptor, materializedExchangeSource, statsAndCosts, jsonRepresentation);
+        return new PlanFragment(id, root, symbols, partitioning, tableScanSchedulingOrder, partitioningScheme.withBucketToPartition(bucketToPartition), stageExecutionDescriptor, outputTableWriterFragment, statsAndCosts, jsonRepresentation);
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java]: Every-time I see this,  I want to make it one parameter per line :P

Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java]: - What do we say to the God of the refactoring?
- Not today

😄

Comment (arhimondr) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java]: Let me open a separate PR with the reformatting

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java [hunk 2]
```diff
@@ -312,6 +324,17 @@ private SubPlan buildFragment(PlanNode root, FragmentProperties properties, Plan
 
             Map<Symbol, Type> fragmentSymbolTypes = filterKeys(symbolAllocator.getTypes().allTypes(), in(extractOutputSymbols(root)));
             planSanityChecker.validatePlanFragment(root, session, metadata, sqlParser, TypeProvider.viewOf(fragmentSymbolTypes), warningCollector);
+
+            Set<PlanNodeId> tableWriterNodeIds = getTableWriterNodeIds(root);
+            boolean outputTableWriterFragment = tableWriterNodeIds.stream().anyMatch(outputTableWriterNodeIds::contains);
+            if (outputTableWriterFragment) {
+                verify(
+                        outputTableWriterNodeIds.containsAll(tableWriterNodeIds),
+                        "outputTableWriterNodeIds %s must include all tableWriterNodeIds %s",
```
Comment (wenleix) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java]: nit: "outputTableWriterNodeIds %s must include either all or none of tableWriterNodeIds %s"

File: presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java (hunk 1 above)
Comment (wenleix) [reply presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragment.java]: Actually I did this in https://github.com/prestodb/presto/pull/12869 :)

**Pull request merged**